pkg/skaffold/deploy/status_check.go

```diff
@@ -212,9 +217,35 @@ func newCounter(i int) *counter {
 	}
 }
 
-func (c *counter) markProcessed(err error) int {
+func (c *counter) markProcessed(err error) *counter {
 	if err != nil {
 		atomic.AddInt32(&c.failed, 1)
 	}
-	return int(atomic.AddInt32(&c.pending, -1))
+	atomic.AddInt32(&c.pending, -1)
+	return c.copy()
+}
+
+func (c *counter) copy() *counter {
```

Comment:
why are we returning a pointer here? 
I think by value should be sufficient - we don't want to sound like that we are returning a mutable structure here -  also  `c == nil` will never evaluate to true!!

Reply:
make sense. Copy should return a non mutable value.